src/main/java/com/project/service/MyPageService.java

```diff
@@ -28,30 +29,35 @@ public class MyPageService {
     private final RankingDayRepository rankingDayRepository;
     private final MyPageMapper myPageMapper; // Mapper 주입
 
+
     public MyPageResponse getMyPage(Long userId, int year, int month, String dateStr) {
 
-        // 1. 데이터 조회 및 계산
         // 1. 유저 조회
         UserEntity user = userRepository.findById(userId)
                         .orElseThrow(() -> new BusinessException(ErrorCode.USER_NOT_FOUND));
 
-        // 2. 기준 날짜 설정 & 유효성 검증
-        LocalDate targetDate = determineTargetDate(year, month, dateStr);
+        // 제외할 시간대 계산 (가입 직후 첫 배치 시간)
+        // 가입: 13:30 -> 제외 범위 : 14:00:00 ~ 14:59:59
+        LocalDateTime createdAt = user.getCreatedAt();
+        LocalDateTime ignoreStart = createdAt.plusHours(1).withMinute(0).withSecond(0).withNano(0);
+        LocalDateTime ignoreEnd = ignoreStart.withMinute(59).withSecond(59);
```

Comment:
@rettooo 님이 작성하신 부분도 로직으로는 이상없다고 판단됩니다. 하지만, ignoreEnd부분에서 나노초 단위까지 처리가 되지 않아 초 단위 아래에서는 조건이 부정확하다고 생각합니다.
좀 더 정확한 시간 계산을 위해서 아래와 같은 방식은 어떤가요?
```suggestion
//시간 차이
Duration diff = Duration.between(유저 createdAt, 문제 solved_time);
//범위에 적합한가 -> true:적합 / false:부적합
// 13:30분 회원 가입 , 15:00분 배치부터 랭킹 집계해야함. -> (15:00 - 13:30)=1시간 30분 >= 2시간 만족못하므로 집계제외
//13:01분 회원 가입 , 15:00분 배치부터 랭킹 집계해야함 -> (15:00 - 13:01) = 1시간 59분 >= 2시간 만족못하므로 집계제외
//회원가입 이후 배치가 돌아가므로 유저 createdAt가 더 작은 경우는 생기지 않아 diff값이 음수되는 경우는 처리하지 않아도됨.
boolean fit = diff.compareTo(Duration.ofHours(1))>=2;

```